Arm64 support

ahossny opened the pull request:
Adds Support to ARM64
Migrated to :NET8

I have started this project and my aim to add arm64 support, as it would be more efficient and would not drain my battery. 

<img width="913" alt="image" src="https://github.com/user-attachments/assets/23f569d2-4d12-4134-9738-23d56436a318" />

ahossny commented:
All build failed as build pipelines requires to be updated, and use Windows SDK 10

ahossny commented:
All build pipelines need to be updated, they should include Windows SDK 10 in order in order to resolve the issue.

GriffinCodes commented:
Would you be able to share the build so I could use and test this? Greatly appreciate your work!

ahossny commented:
@GriffinCodes, I have added a release for arm64 here [https://github.com/ahossny/ShareX/releases/tag/18.0.0-b1](url). Let me know if you find any issues.

SemyonL commented:
Hi @ahossny,
installed the build, but it says dotnet runtime is missing. 
![Screenshot 2025-04-03 185825](https://github.com/user-attachments/assets/6d8bc3d5-f526-4a57-907e-c1779fe00520)
![Screenshot 2025-04-03 190753](https://github.com/user-attachments/assets/e11634ae-27fa-4613-9bb0-e9b2d6170ca3)

SemyonL commented:
@ahossny,
A ShareX.runtimeconfig.json right next to exe is needed with this content:
```javascript
{
    "runtimeOptions": {
      "tfm": "net8.0",
      "framework": {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "8.0.0"
      }
    }
}
``` 

Then the app runs normally. So far it works great, thank you for your work!

ahossny commented:
@SemyonL, thank you for your feedback. I have updated the code to contain this configuration file. and a new build is now published.
[Release](https://github.com/ahossny/ShareX/releases)

ahossny commented:
Develop branch already upgraded to .NET9, replaced previous implementation with what already available in develop branch. Added InnoSetup script to build ARM64 native binary.

Unofficial build available [here](https://github.com/ahossny/ShareX/releases/)
